haystack/components/rankers/diversity.py

```diff
@@ -0,0 +1,215 @@
+import logging
+from typing import Any, Dict, List, Literal, Optional
+
+from haystack import ComponentError, Document, component, default_from_dict, default_to_dict
+from haystack.lazy_imports import LazyImport
+from haystack.utils import ComponentDevice, Secret, deserialize_secrets_inplace
+
+logger = logging.getLogger(__name__)
+
+
+with LazyImport(message="Run 'pip install \"sentence-transformers>=2.2.0\"'") as torch_and_transformers_import:
+    import torch
+    from sentence_transformers import SentenceTransformer
+
+
+@component
+class DiversityRanker:
+    """
+    Implements a document ranking algorithm that orders documents in such a way as to maximize the overall diversity
+    of the documents.
+    It uses a pre-trained Sentence Transformers model to embed the query and the Documents.
+
+    Usage example:
+    ```python
+    from haystack import Document
+    from haystack.components.rankers import DiversityRanker
+
+    ranker = DiversityRanker(model="sentence-transformers/all-MiniLM-L6-v2", similarity="cosine")
+    docs = [Document(content="Paris"), Document(content="Berlin")]
+    query = "What is the capital of germany?"
+    output = ranker.run(query=query, documents=docs)
+    docs = output["documents"]
+    assert len(docs) == 2
+    assert docs[0].content == "Paris"
+    ```
+    """
+
+    def __init__(
+        self,
+        model: str = "sentence-transformers/all-MiniLM-L6-v2",
+        top_k: int = 10,
+        device: Optional[ComponentDevice] = None,
+        token: Optional[Secret] = Secret.from_env_var("HF_API_TOKEN", strict=False),
+        similarity: Literal["dot_product", "cosine"] = "dot_product",
+        prefix: str = "",
+        suffix: str = "",
```

Comment:
I think it would be great if we could expand and update these prefix and suffixes to work for both the query and documents separately. For example, like in our TransformersSimilarityRanker https://github.com/deepset-ai/haystack/blob/0aed48c1c55b557d4a1e7629f4e7fbe7fe72671e/haystack/components/rankers/transformers_similarity.py#L47-L48
since it's possible the chosen needs a different prefix for a query vs a passage.

Reply:
Makes sense, I will add update it to have suffixes and prefixes for both the query and documents separately.